### ISSUE TITLE
GKL fails in Mac

### ISSUE THREAD

**Issue by magicDGS** [broadinstitute/gatk#2315]
## System
* Mac OS X 10.11.6 x86_64
* Java HotSpot(TM) 64-Bit Server VM 1.8.0_60-b27

## Problem
I'm trying to update my project ([ReadTools](https://github.com/magicDGS/ReadTools)) to the latest version of GATK and this dependency throws the following error with some of my gradle tests and while running an uber-jar (using `--use_jdk_deflater false`):

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x000000011d925644, pid=7088, tid=20739
#
# JRE version: Java(TM) SE Runtime Environment (8.0_60-b27) (build 1.8.0_60-b27)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.60-b23 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libgkl_compression8215566221555962564.dylib+0x1644]  Java_com_intel_gkl_compression_IntelDeflater_resetNative+0x164
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/daniel/workspaces/ReadTools/hs_err_pid7088.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

Find attached the log: [hs_err_pid7088.log.txt](https://github.com/broadinstitute/gatk/files/652421/hs_err_pid7088.log.txt)

**Comment by magicDGS** [comment]
I can't assign anyone, @droazen. Could you do it?

**Comment by droazen** [comment]
For @gspowley

**Comment by magicDGS** [comment]
More information: version 0.2.0 is working.

**Comment by gspowley** [comment]
@magicDGS, is this the same Mac as the one in #1985? Assuming yes, this crash happens because AVX is not supported. We had a check for AVX in GKL 0.2.0, but the call was removed by mistake in 0.3.0. 

Sorry about that, we'll fix it in the next GKL release.

In the meantime, you can use the workaround from #2302 and define this environment variable: `export GKL_USE_LIB_PATH=1`

**Comment by magicDGS** [comment]
It is th esame computer, @gspowley. In the meanwhile I will change the dependency to 0.2.0 in my software, because it is mostly used in this kind of system. Also with `--use_jdk_deflater true` it will work.

Thanks for the help!

**Comment by lbergelson** [comment]
@gspowley Would it be possible to add an osx build to the gkl test runs? Likewise, we need an osx run on our travis runs.

**Comment by gspowley** [comment]
We do test GKL on Mac/OSX, but not on a Mac without AVX...

**Comment by lbergelson** [comment]
Oh, good point.  I'm not sure that there is any way we could set up a cloud test for that without buying an old mac putting it in a datacenter somewhere ourself.

**Comment by magicDGS** [comment]
Is there any progress on this?

**Comment by lbergelson** [comment]
A GKL update is coming very soon.  As far as I know it should address this problem.

**Comment by gspowley** [comment]
Right, we're waiting on an HTSJDK release before the next GKL release. After that, we'll add the new GKL to GATK.

**Comment by magicDGS** [comment]
Thanks for the feedback!

**Comment by magicDGS** [comment]
Thanks @gspowley! I test the current master and it's working perfectly!

**Comment by gspowley** [comment]
Great! Thanks for the feedback @magicDGS.
